Docs/sphinx_documentation/source/External_Profiling_Tools.rst

```diff
@@ -279,3 +279,167 @@ generated figures. Some examples are shown here.
+
+
+Nsight Compute
+==============
+
+The NSight Compute tool provides a detailed, fine-grained analysis of your CUDA kernels, 
+giving details about the kernel launch, occupancy, and limitations while suggesting possible
+improvements to maximize the use of the GPU.  It analyzes CUDA-based GPU codes and is available
+on Summit and Cori in system modules.
+
+NSight Compute provides a variety of profiling options.  This documentation will focus on the
+most commonly used options for AMReX users, primarily to keep track of useful flags and analysis
+patterns.  For the complete details of using NSight Compute, refer to the official documentation.
+
+
+Kernel Analysis
+---------------
+
+The standard way to run NSight Compute on an AMReX application is to specify an output file
+that will be transferred to a local workstation of machine for viewing in the NSight Compute GUI.
+NSight Compute can be told to return a report file using the ``-o`` flag. In addition, when
+running with NSight compute on an AMReX application, it is important to turn off the floating
+point exception trap, as it causes a runtime error.  So, an entire AMReX application can be 
+analyzed with NSight Compute by running:
+
+.. highlight:: c++
+
+    ncu -o <file_name> ${EXE} ${INPUTS} amrex.fpe_trap_invalid=0
+
+However, this implementation should almost never used by AMReX applications, as the analysis of
+every kernel would be  extremely lengthy and unnecessary.  To analyze a desired subset of CUDA
+kernels, AMReX users can use the Tiny Profiler's built-in NVTX regions to narrow the scope of
+the analysis.  NSight Compute allows users to specify which NVTX regions to include and exclude
+through the ``--nvtx``, ``--nvtx-include`` and ``--nvtx-exclude`` flags. For example:
+
+.. highlight:: c++
+
+    ncu --nvtx --nvtx-include "Hydro()" --nvtx-exclude "StencilA(),StencilC()" -o kernels ${EXE} ${INPUTS} amrex.fpe_trap_invalid=0
+
+will return a file named ``kernels`` which contains an analysis of the CUDA kernels launched inside
+the ``Hydro()`` region, ignoring any kernels launched inside ``StencilA()`` and ``StencilC()``.  
+When using the NVTX regions built into AMReX's TinyProfiler, be aware that the application must be built
+with ``TINY_PROFILE=TRUE`` and the NVTX region names are identical to the TinyProfiler timer names.
+
+Another helpful flag for selecting a reasonable subset of kernels for analysis is the ``-c`` option. This
+flag specifies the total number of kernels to be analyzed. For example:
+
+ .. highlight:: c++
+
+    ncu --nvtx --nvtx-include "GravitySolve()" -c 10 -o kernels ${EXE} ${INPUTS} amrex.fpe_trap_invalid=0
+
+will only analyze the first ten kernels inside of the ``GravitySolve()`` NVTX region.
+
+For further details on how to choose a subset of CUDA kernels to analyze, or to run a more detailed
+analysis, including CUDA hardware counters, refer to the NSight Compute official documentation.
```

Comment:
You might want to link directly to NVIDIA documentation in the places where you mention it.

Reply:
I was considering this. But, do we want to maintain potentially changing documentation links? This issue comes up a lot and different groups have their own preference.

If no one minds (e.g. @WeiqunZhang, @atmyers), I'm game to put in the links. It's just require a little more maintenance.

Reply:
Fixing broken links is one way we can continue inflating our PR count :-)